### ISSUE TITLE
Ability for RibbonUI to load different icon sets 

### ISSUE THREAD

**Issue by luzpaz** [geolta/FreeCAD-Ribbon#2]
![image](https://github.com/geolta/FreeCAD-Ribbon/assets/4140247/ad347c44-8157-467f-b7f0-e3274aeceb22)

CC @obelisk79

**Comment by geolta** [comment]
This should be possible the same way as when you use the classic toolbars since the icons are just taken from the commands.

But I don't know how to install the icons of @obelisk79.

**Comment by obelisk79** [comment]
You need the  .rcc file from here: https://github.com/obelisk79/FreeCAD-Flat-Icons
You also need to have the Icon Themes addon installed from the addon manager. In the readme/github for that it explains where to put the file. I know that on linux it goes in ~/.local/share/FreeCAD/Gui/Icons

Take note, this version of these icons is currently only optimized for use against dark backgrounds/themes. They will either be invisible or just look really bad/incomplete against lighter colors. I'm not sure if your work on ribbon bar is customizable on that level yet.

**Comment by luzpaz** [comment]
@geolta any interest in trying to implement this. I predict that this addon would be wildly more popular if you do.

**Comment by geolta** [comment]
Here is a screenshot of what happens when you simply install both the icons from Obelisk and the Ribbon addon.
The white background is probably due to some QSS or so in the Ribbon Library I use – I'll have a look into that. Thanks for the hint :)

![grafik](https://github.com/geolta/FreeCAD-Ribbon/assets/115890454/b19c6a41-a60c-4334-9522-7258557259ca)

**Comment by geolta** [comment]
And it can be seen, that my scaled versions for the large icons overwrite the icons of obelisk. My best idea to solve such things is that the default icons from FreeCAD should be a icon theme with a rcc file as well so that the scaled versions could be bundled together.

**Comment by obelisk79** [comment]
Couldn't you scale via code since the icons are SVG based?

Edit: also, is the ribbon customizable at all? There's been some recent changes in the dev builds which would make this potentially more compact and allow for only two rows of icons (in sketcher at least)

**Comment by geolta** [comment]
> Couldn't you scale via code since the icons are SVG based?

Unfortunately not. As outlined in this issue (https://github.com/FreeCAD/FreeCAD/issues/10641#issuecomment-1715505063), simple scaling does work but this would change line thickness as well.

**Comment by geolta** [comment]
> Edit: also, is the ribbon customizable at all? There's been some recent changes in the dev builds which would make this potentially more compact and allow for only two rows of icons (in sketcher at least)

Currently not really. But I'm quite sure that a couple of changes to the pyqtribbon library would allow this. It's a bit of a problem that this addon was not really meant to be used regularly but only to check for technical limitations of the addon approach (this addon is not even in the addon manager).

Currently I don't have the time to put much work into this but if someone wants to, don't hesitate ;)

**Comment by geolta** [comment]
It was indeed one line of QSS that I changed and it is working now :)

The library is now included so there are no additional dependencies (I already updated the instructions in the README). I'm not sure if I did everything right considering the license stuff, I would be very glad if you tell me if I did something wrong.

![grafik](https://github.com/geolta/FreeCAD-Ribbon/assets/115890454/2b7b20de-2e9c-4e47-aa7c-581a15d6b09e)

**Comment by obelisk79** [comment]
I got it installed over here, awesome stuff. It's a shame they don't scale the icons smaller.

**Comment by luzpaz** [comment]
Is there a way we can scale them via the addon instead of tinkering with FreeCAD main branch ?

**Comment by geolta** [comment]
Yes, by default, this is not doable. But I managed to change that :)

When I have it finished, I'll upload it.

The library seems to use a lot of fixed height instead of layout managers, probably it would make sense to change that.

![grafik](https://github.com/geolta/FreeCAD-Ribbon/assets/115890454/d876d683-c24a-444b-8bb4-b8eee799bfe4)

**Comment by geolta** [comment]
Works now out of the box. It simply reads the setting from FreeCAD. But It required some changes in the pyqtribbon library.

And the title bar is still unaffected by this change. I guess this comes at least partially from the OpenDark theme with the large padding.

**Comment by obelisk79** [comment]
@geolta thanks, I'll check this out later today and play around with layout a bit and whatnot
